src/ucp/wireup/wireup_ep.c

```diff
@@ -476,21 +476,24 @@ static UCS_CLASS_CLEANUP_FUNC(ucp_wireup_ep_t)
     uct_worker_progress_unregister_safe(worker->uct, &self->progress_id);
 
     if (self->aux_ep != NULL) {
-        /* No pending operations should be scheduled */
-        uct_ep_pending_purge(self->aux_ep, ucp_destroyed_ep_pending_purge,
-                             ucp_ep);
-        ucp_ep_unprogress_uct_ep(ucp_ep, self->aux_ep, self->aux_rsc_index);
-        uct_ep_destroy(self->aux_ep);
+        /* Discard AUX UCT EP to purge all outstanding/pending operations.
+         * Normally, WIREUP EP should complete all outstanding operations prior
+         * destroying WIREUP EP - so, doing flush(CANCEL) won't take any affect,
+         * but it will make sure that no completions will be received if some
+         * error was detected.
+         * No pending operations should be scheduled */
+        ucp_wireup_ep_discard_aux_ep(self, UCT_FLUSH_FLAG_CANCEL,
+                                     ucp_destroyed_ep_pending_purge, ucp_ep);
         self->aux_ep = NULL;
     }
 
     if (self->super.is_owner && (self->super.uct_ep != NULL)) {
         /* No pending operations should be scheduled */
-        uct_ep_pending_purge(self->super.uct_ep,
-                             ucp_destroyed_ep_pending_purge, ucp_ep);
-        ucp_ep_unprogress_uct_ep(ucp_ep, self->super.uct_ep,
-                                 self->super.rsc_index);
-        uct_ep_destroy(self->super.uct_ep);
+        ucp_worker_discard_uct_ep(self->super.ucp_ep, self->super.uct_ep,
```

Comment:
shall we update the comment in line 492?

Reply:
no, since we still expect that no pending operation will be scheduled, since passing `ucp_destroyed_ep_pending_purge`